service/resources/helpers/helpers.py

```diff
@@ -50,6 +50,11 @@ def get_emails(emails, email_type):
         email_list = []
         counter = 0
         for email in emails:
+            if email_type != 'to' and \
+               'email' not in email or 'name' not in email:
+                print(f'Missing "email" or "name" for email of type "{email_type}" - skipping: {email}')
+                continue
```

Comment:
Skipping it allows us to treat it optionally. If we throw an error, then an email won't be sent, which is the same problem we've been experiencing before. If we want to enforce that, though, it means we should instead change how we pass on data from Form.io's side such that the `CC` is never sent when it is empty (which I think we didn't want to do and preferred to handle it on the service side and make it more resilient?). And, if we want to make this throw an error... I'd say that's only marginally better in the outcome as we now have a more detailed exception vs it just bailing when it tries to access the field and then failing when it's not found.

If `name` is missing, then we could just make it the same as the email as you suggested (since I'm just copying it to the name field anyways on Form.io's side), but it has the side effect of displaying it as `derek.chinn@sfgov.com<derek.chinn@sfgov.com>` instead of just as `derek.chinn@sfgov.com`... so I'd prefer it to just have no name, as it's redundant (but it's not a huge deal IMO).

Reply:
i think optional is good. but as it is now, it will actually skip if someone passes in just the email address. maybe the logic should be skipping when 'email' **and** 'name' is empty?

Reply:
Yea - that was just to handle the way it was coded currently, where it assumes there will always be a email + name.

How about:

* if no e-mail addr (and with no name)
  * skip, print message
* if email addr
  * if no name
    * name = None
* send email to addr + name

Reply:
🚢

Reply:
Addressed in Addressed in [commit](https://github.com/SFDigitalServices/email-microservice-py/pull/21/commits/42194d533ca357a47ae73b29965cbbcb7513a365).